File: Python/numbertree.py
def main():
    s = input().split(' ')
    root = 2**(int(s[0])+1) - 1

    if len(s) == 1:
        print(root)
    else:
        node = 0
        for c in s[1]:
            node *= 2
            node += 1 if c == 'L' else 2

        print(root - node)


if __name__ == "__main__":
    main()

File: Kotlin/numbertree.kt
fun main() {
    val dir = readLine()!!.split(' ')
    val root = Math.pow(2.0, dir[0].toDouble() + 1.0).toInt() - 1
    var node = 0

    if (dir.size == 1) {
        println(root)
        return
    }

    for (c in dir[1]) {
        node *= 2
        node += if (c == 'L') 1 else 2
    }

    println(root - node)
}